testsuite/features/support/commonlib.rb

```diff
@@ -114,7 +114,7 @@ def format_detail(message, last_result, report_result)
 def click_button_and_wait(locator = nil, **options)
   click_button(locator, options)
   begin
-    raise 'Timeout: Waiting AJAX transition (click link)' unless has_no_css?('.senna-loading', wait: 1)
+    raise 'Timeout: Waiting AJAX transition (click link)' unless has_no_css?('.senna-loading', wait: 5)
```

Comment:
5 seconds is too much! It will wait for that for every click that has not .senna-loading in place.
Did you tested in another environment? It might be a big increase in time.

Reply:
It is a "max" value. If it is finished in 0.5 seconds it will continue.

Reply:
It is supposed to be a **max limit** value, it will not wait for every, but only for those that are (for any unknown reason) slow. In the end, in the real life we would wait for that, so we should not say "the feature is failing" but I see better to have a longer running testsuite that tells us what is really failing, then once everything is fixed we talk about **performance** where we have the "Timeout....." error.
At the moment I am afraid we are having masked failures with misleading error messages. I'd like to let them to be discovered (even with a longer run) and then fix them properly.

Reply:
Notice the `has_no` negation, this changes the regular behavior. This waits until the element is not present.
What we need to change is not that timeout here, but the timeout of the click method, the one is injected on `options` in he click_button(locator, options), that's the one we are tweaking when we change the default value `default_max_wait_time`

Reply:
@srbarrios I see the `has_no` negation but what it checks for is that he wants to be sure the async call is already returned. As soon as the AJAX call returns, it continues.
Let me clarify one thing: the `.senna-loading` is the CSS class that only the **green loading bar** DOM has, the one you can notice when SPA is enabled and it is shown during the navigation. It behaves like the following:
- `.senna-loading` is not present
- click on a link
- `.senna-loading` appears
- `.senna-loading` stays there until the AJAX gets the response
- `.senna-loading` disappears as soon as the AJAX gets the response

This is the behavior we really want to mimic, so waiting for the AJAX call to get the response is something we **have to** do if we want to see the content of the page changed, because the AJAX response contains the next content we want to render. So, in the end:

1. `has_no_css?('.senna-loading', wait: 5)` is `true`
2. click on a link
3. while AJAX call does not receive the response, `has_no_css?('.senna-loading')` is `false`
4. **as soon as** AJAX call receives the response, `has_no_css?('.senna-loading')` is `true`
5. if from 3. to 4. it elapses more than 5 seconds, we raise the `exception`

Am I missing something?

Reply:
That's correct. Thinking it better, it will not affect on pages without SPA. It seems safe to merge.
Worst case, we will see results soon.